src/Core/MailTemplate/Transformation/HTMLTextifyTransformation.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\MailTemplate\Transformation;
+
+use Html2Text\Html2Text;
+
+class HTMLTextifyTransformation extends AbstractMailTemplateTransformation
+{
+    /**
+     * @inheritDoc
+     */
+    public function apply($templateContent, array $templateVariables)
+    {
+        $textifier = new Html2Text($templateContent);
```

Comment:
Not sure it's a good idea to transform the html into text when we should be able to manage both versions separately.

Reply:
well actually if a txt template is defined it will be used (the HTMLTextifyTransformation will still be applied though to ensure we don't have HTML code)
but if no txt template is defined then the template used to generate it is the html one (the same fallback exists the other way, if no html template is present the txt one is used form html content)